ast/functions.go

```diff
@@ -467,6 +469,32 @@ func (n *AggregateFuncExpr) updateFirstRow() error {
 	return nil
 }
 
+func (n *AggregateFuncExpr) updateSum() error {
+	ctx := n.GetContext()
+	a := n.Args[0]
+	value := a.GetValue()
+	if value == nil {
+		return nil
+	}
+	if n.Distinct {
+		d, err := ctx.distinctChecker.Check([]interface{}{value})
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if !d {
+			return nil
+		}
+	}
+	sum := ctx.Value
+	var err error
+	sum, err = types.CalculateSum(sum, value)
```

Comment:
Seems we can directly use `ctx.Value, err = types.CalculateSum(ctx.Value, value)`?